modules/main/src/mapbox-legacy/types/style-spec.ts

```diff
@@ -1,77 +1,34 @@
 /*
  * Mapbox Style Specification types
  */
-// Layers
-import type {
-  BackgroundLayer,
-  SkyLayer,
-  CircleLayer,
-  FillLayer,
-  FillExtrusionLayer,
-  HeatmapLayer,
-  HillshadeLayer,
-  LineLayer,
-  RasterLayer,
-  SymbolLayer
-} from 'mapbox-gl';
-
-export type AnyLayer =
-  | BackgroundLayer
-  | CircleLayer
-  | FillExtrusionLayer
-  | FillLayer
-  | HeatmapLayer
-  | HillshadeLayer
-  | LineLayer
-  | RasterLayer
-  | SymbolLayer
-  | SkyLayer;
-
 export type {
-  BackgroundLayer,
-  SkyLayer,
-  CircleLayer,
-  FillLayer,
-  FillExtrusionLayer,
-  HeatmapLayer,
-  HillshadeLayer,
-  LineLayer,
-  RasterLayer,
-  SymbolLayer
-};
-
-// Sources
-import type {
-  Projection as ProjectionSpecification,
-  GeoJSONSourceRaw as GeoJSONSource,
-  VideoSourceRaw as VideoSource,
-  ImageSourceRaw as ImageSource,
-  VectorSource,
-  RasterSource,
-  CanvasSourceRaw as CanvasSource,
-  RasterDemSource
-} from 'mapbox-gl';
+  // Layers
+  AnyLayer as LayerSpecification,
+  BackgroundLayer as BackgroundLayerSpecification,
+  SkyLayer as SkyLayerSpecification,
+  CircleLayer as CircleLayerSpecification,
+  FillLayer as FillLayerSpecification,
+  FillExtrusionLayer as FillExtrusionLayerSpecification,
+  HeatmapLayer as HeatmapLayerSpecification,
+  HillshadeLayer as HillshadeLayerSpecification,
+  LineLayer as LineLayerSpecification,
+  RasterLayer as RasterLayerSpecification,
+  SymbolLayer as SymbolLayerSpecification,
 
-export type AnySource =
-  | GeoJSONSource
-  | VideoSource
-  | ImageSource
-  | CanvasSource
-  | VectorSource
-  | RasterSource
-  | RasterDemSource;
+  // Sources
+  AnySourceData as SourceSpecification,
+  CanvasSourceRaw as CanvasSourceSpecification,
```

Comment:
Can you help me understand why we're aliasing some of these specifications?

Reply:
This will match type names from `@types/mapbox-gl` (unofficial types) with the latest `mapbox-gl` official type names.